src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/BooleanToStringTransformerTest.php

```diff
@@ -60,11 +60,52 @@ public function testReverseTransformFailsIfInteger()
         $this->transformer->reverseTransform(1);
     }
 
+    /**
+     * @expectedException \Symfony\Component\Form\Exception\TransformationFailedException
+     */
+    public function testReverseTransformFailsIfUnexspectedValue()
+    {
+        $this->transformer->reverseTransform('foobar');
+    }
+
     public function testReverseTransform()
```

Comment:
I would add the arguments `$value` and `$result` to the arguments and add a provider `reverseTransformProvider()` which returns pairs like `array('true', true)`. This provider should contain all the test sets you documented in the UPGRADE file. You can then get rid of the below new methods, which are more or less duplicates of this test.